landing-pages/site/content/en/blog/airflow-2.4.0/index.md

```diff
@@ -0,0 +1,163 @@
+
+For more information on dynamic task mapping, see the new sections of the doc on [Transforming Mapped Data][transforming-mapped-data], [Combining upstream data (aka "zipping")][task-mapping-zip], and [Assigning multiple parameters to a non-TaskFlow operator][expand-kwargs].
+
+[task-mapping-zip]: https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/dynamic-task-mapping.html#combining-upstream-data-aka-zipping
+[transforming-mapped-data]: https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/dynamic-task-mapping.html#transforming-mapped-data
+[expand-kwargs]: https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/dynamic-task-mapping.html#assigning-multiple-parameters-to-a-non-taskflow-operator
+
+## Auto-register DAGs used in a context manager (no more `as dag:` needed)
+
+This one is a small quality of life improvement, and I don't want to admit how many times I forgot the `as dag:`, or worse, had `as dag:` repeated.
+
+```python
+
+with DAG(dag_id="example") as dag:
+  ...
+
+
+@dag
+def dag_maker():
+    ...
+
+
+dag2 = dag_maker()
+```
+
+can become
+
+```python
+
+with DAG(dag_id="example"):
+    ...
+
+
+@dag
+def my_dag():
+    ...
+
+
+my_dag()
+```
+
+If you want to disable the behaviour for any reason, set `auto_register=False` on the DAG:
+
+```python
+# This dag will not be picked up by Airflow as it's not assigned to a variable
+with DAG(dag_id="example", auto_register=False):
+    ...
+```
+
+## Removal of experimental Smart Sensors feature
+
+Smart Sensors were added in Airflow 2.0 and are were deprecated starting from Airflow 2.2 in favor of Deferrable operators. If you are using smart sensors, you will have to switch to using deferrable operators before you can upgrade to Airflow 2.4.
+
+We're sorry to remove this feature (we didn't do it lightly) but to enable us to continue to grow and evolve Airflow we needed to remove this experimental code. We will only do this sort of change in a minor release for features marked as experimental. Any feature that is fully supported will only ever be removed in a major release.
+
+## Additional improvements
+
+With over 650 commits the [full list of features, fixes and changes][release-notes-2.4.0] is too big to go in to here (check out the release notes for a full list), but some noteworthy or interesting small features include:
+
+- Auto-refresh on the home page
+- Add `@task.short_circuit` TaskFlow decorator
+- Add roles delete command to cli (#25854)
+- Add support for `TaskGroup` in `ExternalTaskSensor` (#24902)
+- Add `@task.kubernetes` taskflow decorator (#25663)
+- Add experimental `parsing_context` to enable optimization of Dynamic DAG handling in workers (#25161)
+- Consolidate to one `schedule` param (#25410)
+- Allow showing non-sensitive config values in Admin -> Configuration (rather than all or nothing) (#25346)
+- Operator name separate from class (no more `_PythonDecoratedOperator` when using TaskFlow) (#22834)
```

Comment:
```suggestion
- Add roles delete command to cli
- Add support for `TaskGroup` in `ExternalTaskSensor`
- Add `@task.kubernetes` taskflow decorator
- Add experimental `parsing_context` to enable optimization of Dynamic DAG handling in workers
- Consolidate to one `schedule` param
- Allow showing non-sensitive config values in Admin -> Configuration (rather than all or nothing)
- Operator name separate from class (no more `_PythonDecoratedOperator` when using TaskFlow)
```